controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -431,6 +473,39 @@ func checkHttpMonitorExists(dynatraceApiClient *dynatrace.DynatraceApiClient, ho
 	return false, nil
 }
 
+func getClusterRegion(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (string, error) {
+	if hostedcontrolplane == nil {
+		return "", fmt.Errorf("hostedcontrolplane is nil %v", hostedcontrolplane)
+	}
+
+	clusterRegion := hostedcontrolplane.Spec.Platform.AWS.Region
+	if clusterRegion == "" {
+		return "", fmt.Errorf("aws region is not set in hcp %v", hostedcontrolplane)
+	}
+
+	return clusterRegion, nil
+}
+
+func determineDynatraceClusterRegionName(clusterRegion string, monitorLocationType string) (string, error) {
+	var dynatraceClusterRegionName string
+
+	//public
+	if monitorLocationType == "PublicAndPrivate" {
+		dynatraceClusterRegionName, err := getDynatraceEquivalentClusterRegionName(clusterRegion)
+		if err != nil {
+			return "", fmt.Errorf("error getting dynatraceClusterRegionName: %v", err)
+		}
+		return dynatraceClusterRegionName, nil
+
+		//private
+	} else if monitorLocationType == "Private" {
+		dynatraceClusterRegionName = "backplane"
+		return dynatraceClusterRegionName, nil
+	}
+
+	return "", fmt.Errorf("monitorLocationType '%s' not supported", monitorLocationType)
+}
```

Comment:
```suggestion
func determineDynatraceClusterRegionName(clusterRegion string, monitorLocationType string) (string, error) {
	if monitorLocationType == "PublicAndPrivate" {
		return getDynatraceEquivalentClusterRegionName(clusterRegion)
	} else if monitorLocationType == "Private" {
	  // For "Private" HCPs, we set the region to backplane because ...
		return "backplane", nil
	}

	return "", fmt.Errorf("monitorLocationType '%s' not supported", monitorLocationType)
}
```
I would prefer if we use comments to explain the context behind some logic. E.g. the function is named `determine...RegionName`, so the public case makes sense, but the private one isn't so obvious.

Alternatively, if you like the switch/case syntax better it could also be used here, e.g.:
```go
switch monitorLocationType {
  case "PublicAndPrivate":
    return getDynatraceEquivalentClusterRegionName(clusterRegion)
  case "Private":
    // For "Private" HCPs, we set the region to backplane because ...
		return "backplane", nil
  default:
    return "", fmt.Errorf("monitorLocationType '%s' not supported", monitorLocationType)
}
```

Reply:
Added comments - 
```go
func determineDynatraceClusterRegionName(clusterRegion string, monitorLocationType string) (string, error) {
	//public
	if monitorLocationType == "PublicAndPrivate" {
		return getDynatraceEquivalentClusterRegionName(clusterRegion)
	} else if monitorLocationType == "Private" {
		/*
			For "Private" HCPs, we have one backplane location deployed per dynatrace tenant. E.g. "name": "backplanei03xyz"
			"backplane" is returned from this function and passed to GetLocationEntityIdFromDynatrace function and this location is
			searched for in dynatrace - if strings.Contains(loc.Name, locationName) && loc.Type == "PRIVATE" && loc.Status == "ENABLED".
			Ref: https://issues.redhat.com/browse/OSD-25167
		*/
		return "backplane", nil
	}
	return "", fmt.Errorf("monitorLocationType '%s' not supported", monitorLocationType)
}
```

Reply:
Also added comments here https://github.com/openshift/route-monitor-operator/blob/86d2fbb61c3c6e143256b9ff3d3c604d8fc1f7a0/controllers/hostedcontrolplane/hostedcontrolplane.go#L531 to explain the functions that are being called.